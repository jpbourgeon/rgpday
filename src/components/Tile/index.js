import React from 'react'
import PropTypes from 'prop-types'
import Link from '../Link'
import { withStyles } from '@material-ui/core/styles'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import CardMedia from '@material-ui/core/CardMedia'
import Typography from '@material-ui/core/Typography'
import Grid from '@material-ui/core/Grid'
import classnames from 'classnames'

const styles = theme => ({
  card: {
    display: 'flex',
    height: '100%'
  },
  cardDetails: {
    flex: 1
  },
  cardMedia: {
    width: '25%'
  },
  isActive: {
    background: theme.palette.primary.light
  }
})

const Tile = props => {
  const { title: label, description, image, classes, to, isActive } = { ...props }
  let title = (!to) ? label : (<Link to={to} color='primary'>{label}</Link>)
  return (
    <Grid item xs={12} md={6}>
      <Card className={classnames([
        classes.card,
        (!isActive) ? null : classes.isActive
      ])}>
        <div className={classes.cardDetails}>
          <CardContent>
            <Typography variant='h5'>
              {title}
            </Typography>
            <Typography variant='subtitle1' paragraph>
              {description}
            </Typography>
          </CardContent>
        </div>
        {(image === '') ? null : (
          <CardMedia
            className={classes.cardMedia}
            title={title}
            image={image}
          />
        )}
      </Card>
    </Grid>
  )
}

Tile.propTypes = {
  classes: PropTypes.object.isRequired,
  title: PropTypes.node.isRequired,
  image: PropTypes.string,
  description: PropTypes.node,
  isActive: PropTypes.bool,
  to: PropTypes.string,
  xs: PropTypes.number,
  sm: PropTypes.number,
  md: PropTypes.number,
  lg: PropTypes.number,
  xl: PropTypes.number
}

Tile.defaultProps = {
  description: '',
  to: '',
  image: '',
  isActive: false,
  xs: 12,
  sm: 12,
  md: 6,
  lg: 6,
  xl: 6
}

export default withStyles(styles)(Tile)
