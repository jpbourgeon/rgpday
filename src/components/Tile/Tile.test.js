/* global describe, it, expect */
import React from 'react'
import renderer from 'react-test-renderer'
import Tile from './'
import jssSerializer from 'src/utils/jssSnapshotSerializer'

expect.addSnapshotSerializer(jssSerializer)

describe('The Tile component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Tile />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
