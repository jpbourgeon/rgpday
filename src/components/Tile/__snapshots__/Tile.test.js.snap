// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Tile component should render correctly 1`] = `
<div
  className="item-6 12-45 12-59 6-67"
>
  <div
    className="root-103 elevation1-106 rounded-104 root-102 card-1"
  >
    <div
      className="cardDetails-2"
    >
      <div
        className="root-130"
      >
        <h5
          className="root-131 h5-147"
        />
        <p
          className="root-131 subtitle1-149 paragraph-159"
        >
          
        </p>
      </div>
    </div>
  </div>
</div>
`;
